BioCro/DARPA/upload_params.Rmd

```diff
@@ -15,64 +15,138 @@ library(readxl)
+                               Treatment == "outdoor_5cm" ~ "Donald Danforth Plant Science Center Outdoor",
+                               Treatment == "outdoor_JollyG" ~ "Donald Danforth Plant Science Center Outdoor"),
+         variable = case_when(trait == "vmax" ~ "Vcmax", 
+                           trait == "Rd" ~ "leaf_respiration_rate_m2",
+                           trait == "AQY" ~ "quantum_efficiency",
+                           trait == "theta_lc" ~ "theta", 
+                           trait == "g1BB" ~ "stomatal_slope.BB",
+                           trait == "g0BB" ~ "cuticular_cond"),
+         mean = case_when(trait == "g0BB" ~ round(ud.convert(Value, "mol/m2/s", "umol/m2/s"), 0),
+                          TRUE ~ round(Value, 3)),
+         SE = case_when(trait == "g0BB" ~ round(ud.convert(SE, "mol/m2/s", "umol/m2/s"), 0),
+                          TRUE ~ round(SE, 3))) %>% 
+  select(-ID, -SD, -Date.run, -Genotype, -A_Ci, -A_Qin, -Rd, -Treatment, -Value, -trait) %>% 
+  #spread(Trait, value) %>% 
+  dplyr::rename(local_datetime = date) %>% 
   mutate(leafT = 25, 
-         n = 9999) %>% 
-  replace(is.na(.), "")
+         n = 1,
+         species = "Setaria viridis",
+         cultivar = "ME-034",
+         citation_author = "Zhang, Gehan, LeBauer, Riemer, Tarin, Vargas",
+         citation_year = "2018", 
+         citation_title = "Unpublished DARPA experimental data",
+         access_level = "2" 
+         ) %>%
+  select(citation_author, citation_year, citation_title, 
+         species, cultivar, site, treatment, local_datetime, Date, rep,
+         #Vcmax, leaf_respiration_rate_m2, quantum_efficiency, stomatal_slope.BB, cuticular_cond,
+         variable, mean, 
+         leafT, n, SE, access_level) %>%
+  replace(., is.na(.), "")
 ```
 
-Save cleaned parameter as a .csv file in `upload_params_data` folder in `model-vignettes/BioCro/DARPA`. 
+Add a unique entity comprising of date, treatement, and rep
 ```{r}
-if(!dir.exists("upload_params_data/")){
-  dir.create("upload_params_data")
-}
+upload_parameters$entity <- paste0(upload_parameters$Date, "_", upload_parameters$treatment, "_",
+                                   upload_parameters$rep)
+```
 
+Save cleaned parameter as a .csv file in `upload_params_data` folder in `model-vignettes/BioCro/DARPA`. 
+```{r}
 write.csv(upload_parameters, "upload_params_data/phys_parameters.csv", row.names = FALSE)
 ```
 
-Upload that new parameter data file using the [Bulk Upload](http://welsch.cyverse.org:8000/bety/bulk_upload/start_upload) in Welsch BETYdb. The following additional data, which applies across all the parameters, will have to be entered
 
-- site: Donald Danforth Plant Science Center Growth Chamber
-- species: Setaria viridis
-- access_level: Internal & Collaborators
-- cultivar: ME-034
+Split dataframe, first by new data since last  upload, then by parameter.  
```

Comment:
```suggestion
Split dataframe, first by new data since last  upload, then by parameter.  Uploading to BETYdb using the API requires one file per parameter. 
```

Reply:
done